### ISSUE TITLE
Timeout when starting streaming updater

### ISSUE THREAD

**Issue by gjacobrobertson** [launchdarkly/erlang-server-sdk#21]
**Describe the bug**
Sometimes when starting a streaming LD instance from a process outside of the `ldclient` application, an error is raised in the calling process indicating a timeout. This error is raised independently from the `ldclient` application continuing to successfully start the instance.

**To reproduce**
Invoke `ldclient.start_instance/1` with a valid SDK_KEY under network conditions that would cause starting a streaming updater to take longer than 5 seconds. 

**Expected behavior**
`ldclient.start_instance/3` returns `ok` without raising an error after successfully establishing a streaming connection.

**Logs**
```
[error] Task #PID<0.529.0> started from FeatureFlag.Supervisor terminating
** (stop) exited in: :gen_server.call(#PID<0.565.0>, {:listen})
    ** (EXIT) time out
    (stdlib) gen_server.erl:215: :gen_server.call/2
    (ldclient) /opt/app/deps/ldclient/src/ldclient_instance.erl:119: :ldclient_instance.start_updater/3
    (elixir) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Function: &:ldclient.start_instance/1
    Args: [<MY_SDK_KEY>]
```

**SDK version**
1.0.0-beta2

**Language version, developer tools**
Elixir 1.9.4

**OS/platform**
Docker container running alpine-3.11 linux inside Docker for Mac 2.2.0.5(43884)

**Additional context**
The issue seems to stem from the fact that `gen_server.call/2` has a default timeout of 5 seconds, and `ldclient_update_stream_server.do_listen/1` can take longer than 5 seconds if opening a gun connection times out even once. When `gen_server.call/2` raises an error, that is within the call stack of the process that invoked `ldclient.start_instance/3` in the first place and has no impact on `ldclient_update_stream_server.do_listen/1` continuing with its backoff and retry loop. To ensure that `ldclient.start_instance/3` returns after the updater is listening, and avoid killing the calling application while the updater is still establishing its connection, I think it would solve the problem to update `ldclient_update_stream_server.listen/0` to wait indefinitely when calling the server e.g.

```erlang
-spec listen(Pid :: pid()) ->
    ok.
listen(Pid) ->
    gen_server:call(Pid, {listen}, infinity).
```

**Comment by gwhelanLD** [comment]
Hi @gjacobrobertson,

Thanks for using the beta and giving us a detailed issue report! We'll take a look at this issue and work on getting a fix in.

Thanks!
- @gwhelanLD

**Comment by zurab-darkly** [comment]
@gjacobrobertson this issue is fixed as of 1.0.0-beta4.
